### PR TITLE
fix(datepicker): force Intl.DateTimeFormat to use UTC time zone ...

### LINKED ISSUE #5185: Datepicker: Incorrect date displayed when selecting a date in the first week of November 1999
#### Bug, feature request, or proposal:
Bug

#### What is the expected behavior?
When selecting November, 4, 1999 from the datepicker, the input will display 11/04/1999

#### What is the current behavior?
When selecting November, 4, 1999 from the datepicker the input displays 1 day behind, 11/03/1999

#### What are the steps to reproduce?
Select November 4, 1999 from the datepicker in the supplied plunkr:
http://plnkr.co/edit/4kiAVKRyskrJFxHNZLuy

#### What is the use-case or motivation for changing an existing behavior?
It's a bug, the existing behavior is unintended

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
Not sure, just something I noticed on the Material documentation site.  I'm using Chrome 58.0.3029.110 as my browser 

#### Is there anything else we should know?
This happens for the first week of November in both 1999 and 2000, likely other places as well.

### PR DESCRIPTION
**Pull request by mmalerba**
so it doesn't display incorrect dates.

fixes #5185

@julianobrasil Thanks for tracking down the root issue!

### CODE REVIEW

**Comment by crisbeto** [comment]
@mmalerba seems like there's a linting error and the output format is different on iOS Safari.

**Comment by mmalerba** [comment]
lint + tests fixed

File: src/lib/core/datetime/native-date-adapter.ts [hunk 1]
```diff
@@ -154,6 +162,12 @@ export class NativeDateAdapter extends DateAdapter<Date> {
 
   format(date: Date, displayFormat: Object): string {
     if (SUPPORTS_INTL_API) {
+      if (this.useUtcForDisplay) {
+        date = new Date(Date.UTC(
+            date.getFullYear(), date.getMonth(), date.getDate(), date.getHours(),
+            date.getMinutes(), date.getSeconds(), date.getMilliseconds()));
+        displayFormat = Object.assign({}, displayFormat, {timeZone: 'utc'});
```
Comment (kara) [inline src/lib/core/datetime/native-date-adapter.ts]: `Object.assign` is breaking in Google. Can you switch to our util `extendObject`?

**Comment by mmalerba** [comment]
fixed

**Pull request merged**